zebra-state/src/service/non_finalized_state.rs

```diff
@@ -418,12 +418,10 @@ impl NonFinalizedState {
         chain_push_result.expect("scope has finished")
     }
 
-    /// Returns the length of the non-finalized portion of the current best chain.
-    pub fn best_chain_len(&self) -> u32 {
-        self.best_chain()
-            .expect("only called after inserting a block")
-            .blocks
-            .len() as u32
+    /// Returns the length of the non-finalized portion of the current best chain
+    /// or `None` if the best chain has no blocks.
+    pub fn best_chain_len(&self) -> Option<u32> {
```

Comment:
This signature change was done because `shorter_chain_can_be_best_chain` test panics.